llvm/lib/Target/AMDGPU/AMDGPUTargetTransformInfo.cpp

```diff
@@ -1423,3 +1424,32 @@ void GCNTTIImpl::collectKernelLaunchBounds(
   LB.push_back({"amdgpu-waves-per-eu[0]", WavesPerEU.first});
   LB.push_back({"amdgpu-waves-per-eu[1]", WavesPerEU.second});
 }
+
+InstructionCost GCNTTIImpl::getMemoryOpCost(unsigned Opcode, Type *Src,
+                                            Align Alignment,
+                                            unsigned AddressSpace,
+                                            TTI::TargetCostKind CostKind,
+                                            TTI::OperandValueInfo OpInfo,
+                                            const Instruction *I) {
+  if (VectorType *VecTy = dyn_cast<VectorType>(Src))
```

Comment:
I think the cost should be the same for any illegal type -- so instead of checking for i8 vectors can check for illegal types.

```
 TargetLoweringBase::LegalizeKind LK = TLI->getTypeConversion(Src->getContext(), EVT::getEVT(Src->getScalarType(), false));
 if (LK.first != TargetLoweringBase::TypeLegal) { }
```

Reply:
I would like to keep this patch about i8s only with other types potentially enabled by future patches.